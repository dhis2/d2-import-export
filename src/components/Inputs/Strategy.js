import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { RadioGroupField } from '../index'

const strategyOptions = [
    {
        value: 'NEW_AND_UPDATES',
        label: i18n.t('Import new values and update existing'),
    },
    {
        value: 'NEW',
        label: i18n.t('Import new values only'),
    },
    {
        value: 'UPDATES',
        label: i18n.t('Only update existing values, ignore new values'),
    },
    {
        value: 'DELETE',
        label: i18n.t('DELETE: Remove values included in uploaded file'),
    },
]
const defaultStrategyOption = strategyOptions[0].value

const NAME = 'strategy'
const DATATEST = 'input-strategy'
const LABEL = i18n.t('Strategy')

const Strategy = () => (
    <RadioGroupField
        name={NAME}
        label={LABEL}
        options={strategyOptions}
        dataTest={DATATEST}
        vertical
    />
)

export { Strategy, defaultStrategyOption }
