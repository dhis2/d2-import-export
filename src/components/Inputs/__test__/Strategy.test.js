import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Strategy } from '../Strategy'

jest.mock('react-final-form', () => ({
    useField: jest.fn(() => ({
        input: {
            name: 'Name',
            value: '',
            onChange: jest.fn(),
        },
        meta: {
            touched: false,
            error: '',
        },
    })),
}))

describe('Input component - Strategy', () => {
    it('should render correctly', () => {
        const file = shallow(<Strategy />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
