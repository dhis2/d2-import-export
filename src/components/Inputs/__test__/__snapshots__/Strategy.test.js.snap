// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Strategy should render correctly 1`] = `
<RadioGroup
  label="Strategy"
  name="strategy"
  options={
    Array [
      Object {
        "label": "New and updates",
        "value": "NEW_AND_UPDATES",
      },
      Object {
        "label": "New only",
        "value": "NEW",
      },
      Object {
        "label": "Updates only",
        "value": "UPDATES",
      },
      Object {
        "label": "Delete",
        "value": "DELETE",
      },
    ]
  }
/>
`;
